backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/contract/ModCntr_InvoicingGroup_Product_StepDef.java

```diff
@@ -0,0 +1,147 @@
+import org.compiere.model.I_AD_Message;
+import org.compiere.model.I_M_Product;
+import org.compiere.model.I_ModCntr_InvoicingGroup;
+import org.compiere.model.I_ModCntr_InvoicingGroup_Product;
+import org.compiere.util.Env;
+
+import java.util.Map;
+
+import static de.metas.cucumber.stepdefs.StepDefConstants.TABLECOLUMN_IDENTIFIER;
+import static org.assertj.core.api.Assertions.*;
+import static org.compiere.model.I_AD_Message.COLUMNNAME_AD_Message_ID;
+
+public class ModCntr_InvoicingGroup_Product_StepDef
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IMsgBL msgBL = Services.get(IMsgBL.class);
+
+	private final ModCntr_InvoicingGroup_Product_StepDefData modCntrInvoicingGroupProductTable;
+	private final ModCntr_InvoicingGroup_StepDefData modCntrInvoicingGroupTable;
+	private final M_Product_StepDefData productTable;
+	private final AD_Message_StepDefData messageTable;
+
+	public ModCntr_InvoicingGroup_Product_StepDef(
+			@NonNull final ModCntr_InvoicingGroup_Product_StepDefData modCntrInvoicingGroupProductTable,
+			@NonNull final ModCntr_InvoicingGroup_StepDefData modCntrInvoicingGroupTable,
+			@NonNull final M_Product_StepDefData productTable,
+			@NonNull final AD_Message_StepDefData messageTable)
+	{
+		this.modCntrInvoicingGroupProductTable = modCntrInvoicingGroupProductTable;
+		this.modCntrInvoicingGroupTable = modCntrInvoicingGroupTable;
+		this.productTable = productTable;
+		this.messageTable = messageTable;
+	}
+
+	@Given("metasfresh contains ModCntr_InvoicingGroup_Product:")
+	public void metasfresh_contains_ModCntr_InvoicingGroup_Product(@NonNull final DataTable dataTable)
+	{
+		for (final Map<String, String> row : dataTable.asMaps())
+		{
+			createModCntrInvoicingGroupProduct(row);
+		}
+	}
+
+	@Given("the ModCntr_InvoicingGroup_Product is added expecting error:")
+	public void metasfresh_adds_ModCntr_InvoicingGroup_Product_expectingError(@NonNull final DataTable dataTable)
+	{
+		for (final Map<String, String> row : dataTable.asMaps())
+		{
+			boolean errorThrown = false;
+
+			try
+			{
+				createModCntrInvoicingGroupProduct(row);
+				assertThat(1).as("An Exception should have been thrown !").isEqualTo(2);
+			}
+			catch (final AdempiereException e)
+			{
+				errorThrown = true;
+
+				final String errorMessageIdentifier = DataTableUtil.extractStringOrNullForColumnName(row, "OPT." + COLUMNNAME_AD_Message_ID + "." + TABLECOLUMN_IDENTIFIER);
+				if (Check.isNotBlank(errorMessageIdentifier))
+				{
+					final I_AD_Message errorMessage = messageTable.get(errorMessageIdentifier);
+					assertThat(e.getMessage()).contains(msgBL.getMsg(Env.getCtx(), AdMessageKey.of(errorMessage.getValue())));
+				}
+			}
+
+			assertThat(errorThrown).isTrue();
+		}
+	}
+
+	private void createModCntrInvoicingGroupProduct(@NonNull final Map<String, String> row)
+	{
+		final I_ModCntr_InvoicingGroup_Product modCntrInvoicingGroupProductRecord = CoalesceUtil.coalesceSuppliers(
```

Comment:
```suggestion
		final I_ModCntr_InvoicingGroup_Product modCntrInvoicingGroupProductRecord = CoalesceUtil.coalesceSuppliersNotNull(
```

At this point we're sure this is not null